chrecurringdoncron.php

```diff
@@ -222,22 +219,21 @@ function loadEmailAddressField(contactID)
               });
             });
           }
-      ",
-    ));
-
+        ",
+      ));
+    }
   }
 }
 
-}
-
 function chrecurringdoncron_civicrm_postProcess($formName, &$form) {
  
   if($formName == 'CRM_Contribute_Form_Contribution')
   {
+    //CRM-1448- Make Email field set as Main Primary
     $params = CRM_Utils_Request::exportValues();
     if ($form->getVar('_mode') && $form->getVar('_mode') == 'live') {
       $paymentProcessor = $form->getVar('_paymentProcessor');
-      if(!empty($paymentProcessor['name']) && ($paymentProcessor['name'] == 'Credit Card') && ($params['is_recur'] == 1)){
+      if(!empty($paymentProcessor['name']) && ($paymentProcessor['payment_type'] == 1) && ($params['is_recur'] == 1)){
```

Comment:
Now we have created ExtensionUtils function and predefined payment processor names, by that it will compare payment processor names with currently selected payment processor name and identifies for which payment processor email address field should be mandatory or not.